File: 2022/Day8/Puzzel1.py
class tree:
    def __init__( self, size ):
        self.size = size
        self.visible = False

input = []
with open("input.txt") as inputFile:
    for line in inputFile:
        line = line.strip()
        input.append( [] )
        for Character in line:
            input[-1].append( tree( int(Character) ) )

for y in range(len(input)):
    max = -1
    for x in range(len(input[y])):
        if input[y][x].size > max:
            input[y][x].visible = True
            max = input[y][x].size
    max = -1
    for x in range(len(input[y])):
        if input[y][len(input[y])-x-1].size > max:
            input[y][len(input[y])-x-1].visible = True
            max = input[y][len(input[y])-x-1].size
for x in range(len(input[0])):
    max = -1
    for y in range(len(input)):
        if input[y][x].size > max:
            input[y][x].visible = True
            max = input[y][x].size
    max = -1
    for y in range(len(input)):
        if input[len(input)-y-1][x].size > max:
            input[len(input)-y-1][x].visible = True
            max = input[len(input)-y-1][x].size

count = 0
for line in input:
    for tree in line:
        if tree.visible:
            count += 1
    
print(count)

File: 2022/Day08/Puzzel1.cpp
#include <string>
#include <fstream>
#include <vector>
#include <iostream>

struct tree
{
    int size;
    bool visible;

    tree( int size ):
        size( size ),
        visible( false )
    {}
};

int main()
{
    std::ifstream inputFile("input.txt");
    std::vector<std::vector<tree>> input = {};
    std::string temp;
    int row = 0;
    int amountOfTrees = 0;
    while( std::getline( inputFile, temp )  )
    {
        input.push_back( {} );
        for( char Character: temp )
        {
            input[row].push_back( tree( Character-'0' ) );
            amountOfTrees++;
        }
        row++;
    }
    for( int y=0; y<input.size(); y++ )
    {
        int max=-1;
        for( int x=0; x<input[y].size(); x++)
        {
            if( input[y][x].size > max )
            {
                input[y][x].visible = true;
                max = input[y][x].size;
            }
        }
        max=-1;
        for( int x=input[y].size(); x>0; x--)
        {
            if( input[y][x-1].size > max )
            {
                input[y][x-1].visible = true;
                max = input[y][x-1].size;
            }
        }
    }

    for( int y=0; y<input.size(); y++ )
    {
        int max=-1;
        for( int x=0; x<input[y].size(); x++)
        {
            if( input[x][y].size > max )
            {
                input[x][y].visible = true;
                max = input[x][y].size;
            }
        }
        max=-1;
        for( int x=input[y].size(); x>0; x--)
        {
            if( input[x-1][y].size > max )
            {
                input[x-1][y].visible = true;
                max = input[x-1][y].size;
            }
        }
    }

    int count = 0;
    for( auto line: input )
    {
        for( auto tree: line)
        {
            if( tree.visible )
                count++;
        }
    }
    std::cout << count << '\n';
}